lib/build-tokens.js

```diff
@@ -91,45 +112,62 @@ async function buildTokensCommand(commandArgs) {
         transform: (token) => colorTransform(token, themeVariant),
       },
     },
-    format: {
-      'css/custom-variables': formatterArgs => createCustomCSSVariables({
-        formatterArgs,
-        themeVariant,
-      }),
-    },
     platforms: {
       css: {
         ...coreConfig.platforms.css,
         files: [
           {
             format: 'css/custom-variables',
             destination: `themes/${themeVariant}/variables.css`,
-            filter: hasSourceTokensOnly ? 'isSource' : undefined,
+            filter: hasSourceTokensOnly
+              ? `isSource.${themeVariant}`
+              : `isThemeVariant.${themeVariant}`,
```

Comment:
Just trying to wrap my head around this change a little bit. Before this change if `hasSourceTokensOnly` was `false` we'd have filter as `undefined` - does that mean it used to apply to all variants?

Reply:
Previously, `hasSourceTokensOnly: false` would rely on filtering implemented _within the format function_, instead of as a true `style-dictionary` filter implemented with `.registerFilter`. That is, formatters should assume any token filtering happens _before_ the format is applied, which already limits the contents of `dictionary.allTokens` we previously filtered on explicitly.

There's some more detailed rationale about this change provided on my review of Peter's PR [here](https://github.com/openedx/paragon/pull/3186#discussion_r1739634704), notably:

> IMHO, having `unfilteredTokens` in the `dictionary` here suggests a possible code smell that we might want to be doing such filtering as a legit custom filter registered with `style-dictionary` (similar to how we seem to define an `isSource` filter today via `registerFilter`).

See the current implementation of `createCustomCSSVariables` ([source](https://github.com/openedx/paragon/blob/0e16dbbff83b7e1ee72b93205b193feb3a99603a/tokens/style-dictionary.js#L74-L76)).